### PR TITLE
Fix sneaking bugs (yet again)

### PR DESCRIPTION
**Pull request by sfan5**
(see commit msg)
I was able to locate the cause of #5438 but wasn't successful at fixing it:

The cause is that the sneaking thinks you are trying to sneak on the node below,
this happens when you go directly from the node below onto the snow.
If you "get" onto the snow from above or jump while standing on it sneaking will work 100% reliably.
The code related to this is where `f32 position_y_mod` is defined and set. Also generally this seems to happen only with nodes that have an y-extent < 0.5

### CODE REVIEW

File: src/localplayer.cpp [hunk 1]
```diff
@@ -631,7 +640,7 @@ void LocalPlayer::applyControl(float dtime)
 					speedV.Y = movement_speed_walk;
 			}
 		}
-		else if(m_can_jump || (control.sneak && m_sneak_ladder_detected))
+		else if(m_can_jump)
 		{
```
Comment (nerzhul) [inline src/localplayer.cpp]: can you fix this brace ?

**Comment by paramat** [comment]
:+1: If it hasn't been changed overnight, i tested it and it fixes several issues.

File: src/localplayer.cpp (hunk 1 above)
Comment (paramat) [inline src/localplayer.cpp]: if no-one else does i can fix this on merge.

**Comment by paramat** [comment]
b85301ce69a3dc0763408dafc88cb4c1f56e53cf

File: src/localplayer.cpp (hunk 1 above)
Comment (paramat) [inline src/localplayer.cpp]: Looking at the code, there was no point fixing this here but not in surrounding code, there are so many code style issues they need to be done in another commit.
